variants/github_actions_ci/workflows/ci.yml

```diff
@@ -0,0 +1,157 @@
+name: CI
+on:
+  pull_request:
+  push:
+    branches:
+      - main
+      - production
+concurrency:
+  # Pushing new changes to a branch will cancel any in-progress CI runs
+  group: ${{ github.ref }}
+  cancel-in-progress: true
```

Comment:
@G-Rath This seems to make sense but I haven't lived with it.

Reply:
Yeah neither have I, but it should be a good thing - we have used [`styfle/cancel-workflow-action`](https://github.com/styfle/cancel-workflow-action) on projects though which should be replaceable with this.

Reply:
I'll dogfood it on my current client project.

Reply:
i added it to ripple

it's generally fine as long as deploys won't be cancelled mid deploy

Reply:
@lukeify and I have been using this on a client project and it seems fine so far.